### PR TITLE
Three links

### PR DESCRIPTION
**Pull request by KrasimirRaikov**
Servlet that counts the number of times specific link has been clicked.
counter

### CODE REVIEW

**Comment by panayotkulchev** [comment]
I see that you used mu suggestion to use maven. That is good. Do you figure it out how to set up jetty too?

File: src/test/java/com/clouway/links/CounterTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public void tearDown() throws IOException {
+        writer.close();
+        out.close();
+    }
+
+    @Test
+    public void oneClickOnFirstLink() throws ServletException, IOException {
+
+        context.checking(new Expectations(){{
+            oneOf(request).getSession();will(returnValue(session));
```
Comment (panayotkulchev) [inline src/test/java/com/clouway/links/CounterTest.java]: I see you want to save space but common practice is to spit in two:
```java
oneOf(request).getSession();
will(returnValue(session));
```

File: src/test/java/com/clouway/links/CounterTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            oneOf(session).getId(); will(returnValue("1"));
+            oneOf(request).getParameter("link"); will(returnValue("1"));
+            oneOf(response).setContentType("text/html");
+            oneOf(response).getWriter(); will(returnValue(writer));
+        }});
+        counter.doGet(request, response);
+        writer.flush();
+
+        writer.flush();
```
Comment (panayotkulchev) [inline src/test/java/com/clouway/links/CounterTest.java]: Why you flash twice?

**Comment by KrasimirRaikov** [comment]
I have not used jetty, but I will use it.

File: src/test/java/com/clouway/links/CounterTest.java (hunk 2 above)
Comment (KrasimirRaikov) [inline src/test/java/com/clouway/links/CounterTest.java]: Sorry, I forgot to remove it after i finished experimenting with the test.

**Comment by panayotkulchev** [comment]
If the PR is ready for review do not forget to put the label.

**Comment by KrasimirRaikov** [comment]
I wanted to make some more test scenarios.

File: src/main/java/com/clouway/links/Counter.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        out.write("<body>");
+        out.write("<a href='?link=1'>link1</a><p>count: " + firstCount + "</p></br>");
+        out.write("<a href='?link=2'>link2</a><p>count: " + secondCount + "</p></br>");
+        out.write("<a href='?link=3'>link3</a><p>count: " + thirdCount + "</p></br>");
+        out.write("</body>");
+        out.write("</html>");
+
+    }
+
+    private void setTimesVisited(Map<String, Integer> linkCounter, String sessionId) {
```
Comment (panayotkulchev) [inline src/main/java/com/clouway/links/Counter.java]: Have in mind that in that case you keep more and more data and nothing take care to clean it. Do not change but have it in mind.

**Comment by panayotkulchev** [comment]
LGTM!

**Pull request merged**